src/analytics/formTracker/formTracker.ts

```diff
@@ -168,34 +168,36 @@ export class FormTracker extends BaseTracker {
     const field = document.getElementById(element.id);
     const fieldset = field?.closest("fieldset");
     const checkbox = element.type === "checkbox";
-    const radio = element.type === "radio buttons";
-    if (fieldset) {
-      // If it's a child of a fieldset e.g radio button/ checkbox, look for the legend
-      const legendElement = fieldset.querySelector("legend");
-      if (legendElement?.textContent) {
-        return legendElement.textContent.trim();
-      }
-      // if it is a checkbox or radio which does not have a legend, then check for the below conditions
-    } else if (checkbox || radio) {
-      // Look for h1 or h2 with rel attribute matching element.id
+    const radio = element.type === "radio";
+    const commonId = element.id.split("-")[0];
+
+    const checkCommonSelectors = (): string => {
       const h1OrH2WithRel = document.querySelector(
```

Comment:
What do you think of changing the naming here to something like getHeaderByRelAttribute? or  getHeaderByRelTag...

Reply:
Can take it or leave it, this is a small nitpick

Reply:
I can change but would it be header or heading ?

Reply:
Yes good point, Heading :)